src/test/java/org/folio/rest/impl/ReceivingHistoryTest.java

```diff
@@ -0,0 +1,143 @@
+      testCreatePoLine();
+      testVerifyPoLineCreated();
+
+      logger.info("--- mod-orders-storage receiving_history test: Creating Detail ...");
+      testCreateDetail();
+      testVerifyDetailCreated();
+
+      logger.info("--- mod-orders-storage pieces test: Verify receiving_history view created ...");
+      verifyReceivingHistoryViewCreated(RECEIVING_HISTORY_ENDPOINT);
+
+    } catch (Exception e) {
+      logger.error("--- mod-orders-storage-test: receiving_history API ERROR: " + e.getMessage(), e);
+    } finally {
+      logger.info("--- mod-orders-storage receiving_history test: Clean-up Detail, PoLine and Pieces ...");
+      testDeleteDetail(detailSampleId);
+      testDeletePoLine(poLineSampleId);
+      testDeletePieces(piecesSampleId);
+    }
+  }
+
+  private void testDeleteDetail(String detailSampleId) {
+    deleteData(DETAILS_ENDPOINT, detailSampleId).then().log().ifValidationFails()
+    .statusCode(204);
+  }
+  
+  private void testDeletePoLine(String poLineSampleId) {
+    deleteData(PO_LINE_ENDPOINT, poLineSampleId).then().log().ifValidationFails()
+    .statusCode(204);
+  }
+  
+  private void testDeletePieces(String piecesSampleId) {
+    deleteData(PIECES_ENDPOINT, piecesSampleId).then().log().ifValidationFails()
+    .statusCode(204);
+  }
+  
+  private void testCreateDetail() {
+    String detailSample = getFile("details_for_view.sample");
+    Response detailResponse = postData(DETAILS_ENDPOINT, detailSample);
+    detailSampleId = detailResponse.then().extract().path("id");
+  }
+
+  private void testCreatePiece() {
+    String pieceSample = getFile("pieces_for_view.sample");
+    Response response = postData(PIECES_ENDPOINT, pieceSample);
+    piecesSampleId = response.then().extract().path("id");
+  }
+  
+  private void testCreatePoLine() {
+    String poLineSample = getFile("po_line_for_view.sample");
+    Response response = postData(PO_LINE_ENDPOINT, poLineSample);
+    response.then().log().ifValidationFails()
+      .statusCode(201)
+      .body("description", equalTo("ABCDEFGH"));
+    poLineSampleId = response.then().extract().path("id");
+  }
+  
+  private void testVerifyDetailCreated() {
+    getData(DETAILS_ENDPOINT).then().log().ifValidationFails()
+    .statusCode(200)
+    .body("total_records", equalTo(17));
+  }
+  
+  private void testVerifyPoLineCreated() {
+    getData(PO_LINE_ENDPOINT).then().log().ifValidationFails()
+    .statusCode(200)
+    .body("total_records", equalTo(17));
+  }
+  
+  private void testVerifyPieceCreated() {
+    getData(PIECES_ENDPOINT).then().log().ifValidationFails()
+    .statusCode(200)
+    .body("total_records", equalTo(1));
+  }
+  
+  void verifyViewCollection(String endpoint) {
+    // Verify that there are no existing records in View
+    getViewData(endpoint).then()
+      .log().all()
+      .statusCode(200)
+      .body("total_records", equalTo(0));
+  }
+
+  void verifyReceivingHistoryViewCreated(String endpoint) {
```

Comment:
This method can be combined with the above method if we can take the expected value as parameter, again just for maintainability
ex: verifyViewCollection(String endpoint, int expectedCount)